barretenberg/cpp/src/barretenberg/vm2/constraining/relations/sha256.test.cpp

```diff
@@ -74,5 +90,48 @@ TEST(Sha256ConstrainingTest, Basic)
     check_relation<sha256>(trace);
 }
 
+TEST(Sha256Compression, Interaction)
+{
+    NoopEventEmitter<MemoryEvent> emitter;
+    Memory mem(/*space_id=*/0, emitter);
+    StrictMock<simulation::MockContext> context;
+    EXPECT_CALL(context, get_memory()).WillRepeatedly(ReturnRef(mem));
+
+    EventEmitter<Sha256CompressionEvent> sha256_event_emitter;
+    Sha256 sha256_gadget(sha256_event_emitter);
+
+    std::array<uint32_t, 8> state = { 0, 1, 2, 3, 4, 5, 6, 7 };
+    MemoryAddress state_addr = 0;
+    for (uint32_t i = 0; i < 8; ++i) {
+        mem.set(state_addr + i, state[i], MemoryTag::U32);
+    }
+
+    std::array<uint32_t, 16> input = { 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15 };
+    MemoryAddress input_addr = 8;
+    for (uint32_t i = 0; i < 16; ++i) {
+        mem.set(input_addr + i, input[i], MemoryTag::U32);
+    }
+    MemoryAddress dst_addr = 25;
+
+    sha256_gadget.compression(context, state_addr, input_addr, dst_addr);
+
+    TestTraceContainer trace;
+    tracegen::Sha256TraceBuilder builder(trace);
+    tracegen::PrecomputedTraceBuilder precomputed_builder;
+    // precomputed_builder.process_misc(trace);
+    // Build just enough clk rows for the lookup
+    trace.set(C::precomputed_first_row, 0, 1);
+    for (uint32_t i = 0; i < 65; i++) {
+        trace.set(C::precomputed_clk, i, i);
+    }
```

Comment:
hmm, I'd rather have `PrecomputedTraceBuilder::process_misc(TraceContainer& trace)` take an extra argument `uint32 size = CIRCUIT_SUBGROUP_SIZE` in the header file, which you can change here.